javagen/src/main/java/com/azure/autorest/mapper/ProxyMethodMapper.java

```diff
@@ -177,8 +182,50 @@ protected void buildUnexpectedResponseExceptionTypes(ProxyMethod.Builder builder
                                                          Operation operation, List<HttpResponseStatus> expectedStatusCodes,
                                                          JavaSettings settings) {
         ClassType errorType = null;
+        Map<ClassType, List<HttpResponseStatus>> errorTypeMap = new HashMap<>();
+
+        /*
+        1. If exception has valid numeric status codes, group them to unexpectedResponseExceptionTypes
+        2. If exception does not have status codes, or have 'default' or invalid number, put the first to unexpectedResponseExceptionType, ignore the rest
+        3. After processing, if no model in unexpectedResponseExceptionType, take any from unexpectedResponseExceptionTypes and put it to unexpectedResponseExceptionType
+         */
+
         if (operation.getExceptions() != null && !operation.getExceptions().isEmpty()) {
-            errorType = (ClassType) Mappers.getSchemaMapper().map(operation.getExceptions().get(0).getSchema());
+            //errorType = (ClassType) Mappers.getSchemaMapper().map(operation.getExceptions().get(0).getSchema());
+            for (Response exception : operation.getExceptions()) {
```

Comment:
Does this require additional changes in RestProxy to handle different exception types? Has this been verified with azure-core?

Reply:
I thought I've tested it long time ago for e.g. 404 -> ResourceNotFoundException, 429 -> TooManyRequestException.

I will try it again with httpbin.

Reply:
I've tested it with httpbin, all good.

```
    @Host("{$host}")
    @ServiceInterface(name = "HttpbinService")
    private interface HttpbinService {
        @Post("status/{code}")
        @Headers("accept: application/json")
        @UnexpectedResponseExceptionType(
                value = ResourceNotFoundException.class,
                code = {400, 404})
        @UnexpectedResponseExceptionType(
                value = ResourceModifiedException.class,
                code = {409})
        @UnexpectedResponseExceptionType(HttpResponseException.class)
        Mono<Response<Object>> post(@HostParam("$host") String endpoint,
                                    @PathParam("code") int code,
                                    @QueryParam("queryList") List<String> queryList);
    }
```

exceptions
```
Exception in thread "main" com.azure.core.exception.ResourceNotFoundException: Status code 404

Exception in thread "main" com.azure.core.exception.ResourceModifiedException: Status code 409

Exception in thread "main" com.azure.core.exception.HttpResponseException: Status code 410
```